### ISSUE TITLE
SMS Crashes Freeswitch Cluster

### ISSUE THREAD

**Issue by skyeblueiscool** [fusionpbx/fusionpbx#2306]
We had a user who sent an SMS from a Yealink phone. When he did so it took out 2 of our 3 servers that are in a BDR cluster. Interestingly, the server that the Yealink was registered to did not crash, but the two clustered servers did.

These are the logs from the failed two servers. They are exactly the same. 

2017-01-17 04:49:48.913440 [DEBUG] sofia_presence.c:225 Can't find registered user [cell phone number]@domain
2017-01-17 04:49:48.933436 [INFO] mod_sms.c:337 Processing text message 225->[cell phone number] in context public
2017-01-17 04:49:48.933436 [WARNING] mod_sms.c:366 Context public not found
2017-01-17 04:49:48.933436 [DEBUG] mod_sms.c:446 SMS chatplan no actions found
2017-01-17 04:49:48.933436 [DEBUG] sofia_presence.c:225 Can't find registered user [cell phone number]@domain
2017-01-17 04:49:48.973434 [INFO] mod_sms.c:337 Processing text message 225->[cell phone number] in context public
2017-01-17 04:49:48.973434 [WARNING] mod_sms.c:366 Context public not found
2017-01-17 04:49:48.973434 [DEBUG] mod_sms.c:446 SMS chatplan no actions found
2017-01-17 04:49:48.973434 [DEBUG] sofia_presence.c:225 Can't find registered user [cell phone number]@domain
...skipping...
nbound) =~ /^inbound$/ break=never
9618dcbc-dc70-11e6-9ab3-9ff444179d3c Dialplan: sofia/internal/15038308497@64.136.173.31 Regex (FAIL) [user_record] ${user_record}() =~ /^inbound$/ break=never
9618dcbc-dc70-11e6-9ab3-9ff444179d3c Dialplan: sofia/internal/15038308497@64.136.173.31 Regex (FAIL) [user_record] ${user_exists}(false) =~ /^true$/ break=never
9618dcbc-dc70-11e6-9ab3-9ff444179d3c Dialplan: sofia/internal/15038308497@64.136.173.31 Regex (FAIL) [user_record] ${call_direction}(inbound) =~ /^outbound$/ break=never
9618dcbc-dc70-11e6-9ab3-9ff444179d3c Dialplan: sofia/internal/15038308497@64.136.173.31 Regex (FAIL) [user_record] ${user_record}() =~ /^outbound$/ break=never

**Comment by blackc2004** [comment]
Are you using the fusionpbx-apps/sms?

**Comment by skyeblueiscool** [comment]
No

**Comment by skyeblueiscool** [comment]
Here's some more information.

Our clustered servers are getting this error whenever you receive an sms
2017-01-17 05:15:16.456230 [DEBUG] sofia_presence.c:225 Can't find registered user cellphone@domain
2017-01-17 05:15:16.496230 [DEBUG] sofia_presence.c:225 Can't find registered user cellphone@domain
2017-01-17 05:15:16.516232 [DEBUG] sofia_presence.c:225 Can't find registered user cellphone@domain
2017-01-17 05:15:16.536200 [DEBUG] sofia_presence.c:225 Can't find registered user cellphone@domain
2017-01-17 05:15:16.556292 [DEBUG] sofia_presence.c:225 Can't find registered user cellphone@domain
2017-01-17 05:15:16.576229 [DEBUG] sofia_presence.c:225 Can't find registered user cellphone@domain
2017-01-17 05:15:16.596236 [DEBUG] sofia_presence.c:225 Can't find registered user cellphone@domain
2017-01-17 05:15:16.636230 [DEBUG] sofia_presence.c:225 Can't find registered user cellphone@domain
2017-01-17 05:15:16.656232 [DEBUG] sofia_presence.c:225 Can't find registered user cellphone@domain

**Comment by Len-PGH** [comment]
What version of FreeSWITCH? They just added Native support for Flowroute SMS API https://freeswitch.org/jira/browse/FS-9310 in 1.6.14

**Comment by skyeblueiscool** [comment]
Hey Len,

root@ca1:~# fs_cli -x 'version'
FreeSWITCH Version 1.4.26+git~20160205T175853Z~ca9207aa32~64bit (git ca9207a 2016-02-05 17:58:53Z 64bit)

I'm able to duplicate this reliably every time

**Comment by skyeblueiscool** [comment]
For now I blocked the SMSs by doing this:
-A INPUT -p udp --dport 5060 -m string --string "MESSAGE sip:" --algo bm -j DROP

That should hold us over until we get a proper fix in :)

**Comment by Len-PGH** [comment]
1.4.26 is almost 1 year past EOL.  Something like this almost always would need a ticket opened with FreeSWITCH but they would only look at the ticket if FreeSWITCH was up-to-date.

**Comment by skyeblueiscool** [comment]
So we figured out what was causing this:
The Yealink phone would attempt to send an SMS to Freeswitch. The Freeswitch server would reply 503 - Service Unavailable. This is normal behavior. However, the phone would then "fail over" using our DNS SRV records to the next server in the cluster. When it did that, the secondary server would accept the SMS with a 202 Accepted and then try and route the call. Once it did that, the FS server that attempted to route the call would crash. I've attached screenshots of the SIP transmissions. 

![sip message - failover](https://cloud.githubusercontent.com/assets/13353482/22009439/16a43388-dc38-11e6-946e-a115e5df6169.png)
![sip message - crash](https://cloud.githubusercontent.com/assets/13353482/22009438/16a32d9e-dc38-11e6-939f-53ec7d00aa70.png)
